config/reference_config.go

```diff
@@ -171,6 +172,9 @@ func (refconfig *ReferenceConfig) getUrlMap() url.Values {
 
 	//filter
 	urlMap.Set(constant.REFERENCE_FILTER_KEY, mergeValue(consumerConfig.Filter, refconfig.Filter, constant.DEFAULT_REFERENCE_FILTERS))
+	if refconfig.Generic {
+		urlMap.Set(constant.REFERENCE_FILTER_KEY, mergeValue(consumerConfig.Filter, refconfig.Filter, constant.GENERIC_REFERENCE_FILTERS))
```

Comment:
generic filter should be added to  `DEFAULT_REFERENCE_FILTERS`  instead of replacing it. eg: 
`if refconfig.Generic {
    defaultReferenceFilter = constant.GENERIC_REFERENCE_FILTERS + constant.DEFAULT_REFERENCE_FILTERS
}

urlMap.Set(constant.REFERENCE_FILTER_KEY, mergeValue(consumerConfig.Filter, refconfig.Filter,defaultReferenceFilter  `